src/query/functions/src/scalars/datetime.rs

```diff
@@ -148,6 +149,59 @@ fn register_string_to_timestamp(registry: &mut FunctionRegistry) {
             }
         })(val, ctx)
     }
+
+    registry.register_combine_nullable_2_arg::<StringType, StringType, TimestampType, _, _>(
+        "str_to_timestamp",
+        |_, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_2_arg::<StringType, StringType, NullableType<TimestampType>>(
+            |timestamp, format, output, ctx| {
+                if format.is_empty() {
+                    output.push_null();
+                } else {
+                    match (std::str::from_utf8(timestamp), std::str::from_utf8(format)) {
+                        (Ok(date), Ok(format)) => {
+                            // date need has timezone info.
+                            if let Ok(res) = DateTime::parse_from_str(date, format) {
+                                output.push(
+                                    res.with_timezone(&ctx.func_ctx.tz.tz).timestamp_micros(),
+                                );
+                            } else {
+                                output.push_null();
+                            }
+                        }
+                        _ => {
+                            output.push_null();
+                        }
+                    }
+                }
+            },
+        ),
+    );
+
+    registry.register_combine_nullable_2_arg::<StringType, StringType, DateType, _, _>(
+        "str_to_date",
+        |_, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_2_arg::<StringType, StringType, NullableType<DateType>>(
+            |date, format, output, _| {
+                if format.is_empty() {
+                    output.push_null();
+                } else {
+                    match (std::str::from_utf8(date), std::str::from_utf8(format)) {
+                        (Ok(date), Ok(format)) => {
```

Comment:
why not throw the error?

Reply:
Refer to MySQL behavior here. Return null directly if there is any invalid

Reply:
I think we should not follow MySQL's behavior, we should throw if any error happens.

Reply:
> I think we should not follow MySQL's behavior, we should throw if any error happens.

Modify in this pr: https://github.com/datafuselabs/databend/pull/11455